bin/sync.sh

```diff
@@ -3,27 +3,22 @@
 # `-f` escapes `*`
 set -xf
 
-echo "Before Syncs"
-echo $(date +%s)
+BUCKET_NAME=${BUCKET_PATH:5}
 
 # Sync static assets first, before uploading other files that reference them.
-# Set long live cache headers.
-aws s3 sync ./_site ${BUCKET_PATH} \
-    --exclude "*" \
-    $(jq .paths[] static/staticfiles.json | xargs -I {} echo -n "--include static/{}* ") \
-    --cache-control "max-age=315360000,public,immutable" \
-    --acl public-read \
-    --delete
+# Set long live cache headers using `aws s3api` only to files that changed.
+./bin/rclone --config ./bin/rclone.conf --no-update-modtime -v \
+       $(jq .paths[] static/staticfiles.json | xargs -I {} echo -n "--include static/{} ") \
+       sync ./_site ${BUCKET_PATH} 2>&1 | grep "Copied" | cut -d ":" -f 4 | xargs -I '{}' aws s3api copy-object \
```

Comment:
One thing that could be bad is if the process fails in the middle for some reason, we'll lose the state of which files were changed.

Imagine, one pipeline run changes file a, then the cut/xargs fails for some reason.  pipeline run v2 won't change the file, so the cache-control won't be modified.

We could just break it up into two commands.  Save a file or output to the screen. Then if a failure were to happen it wouldn't be too bad to run the true-up by hand.

Reply:
You have point but I don't think getting the output would provide a good input for manual action. Probably having a cli flag to change the `sync` command to a `copy` command and redo everything is best. Thoughts?

Reply:
Yeah, that sounds good.

Reply:
Added `GLOBAL_SET_METADATA` env variable which controls when scripts set all metadata or only for updated files. Made it a env variable so one can set it when running the pipeline manually.  https://gitlab.com/mozmeao/lumbergh/pipelines/new

https://github.com/mozmeao/lumbergh/pull/518/files#diff-e6b45cef94f317a441d6f606fa6e6d0cR13-R33